certbot/certbot/crypto_util.py

```diff
@@ -173,23 +178,39 @@ def import_csr_file(csrfile, data):
     return PEM, util.CSR(file=csrfile, data=data_pem, form="pem"), domains
 
 
-def make_key(bits):
-    """Generate PEM encoded RSA key.
+def make_key(key_prop):
+    """Generate PEM encoded RSA/EC key.
 
-    :param int bits: Number of bits, at least 1024.
+    :param dict key_prop: Key property such as type, bits, curve name.
 
-    :returns: new RSA key in PEM form with specified number of bits
+    :returns: new RSA/EC key in PEM form with specified number of bits
     :rtype: str
 
     """
-    assert bits >= 1024  # XXX
-    key = crypto.PKey()
-    key.generate_key(crypto.TYPE_RSA, bits)
-    return crypto.dump_privatekey(crypto.FILETYPE_PEM, key)
+    if key_prop["type"] == "rsa":
+        bits = key_prop["size"]
+        assert bits >= 1024  # XXX
+        key = crypto.PKey()
+        key.generate_key(crypto.TYPE_RSA, bits)
+        return crypto.dump_privatekey(crypto.FILETYPE_PEM, key)
+    elif key_prop["type"] == "ecdsa":
+        curve_name = key_prop["curve"]
+        cv = getattr(ec, curve_name.upper())
+        assert curve_name == cv.name
```

Comment:
We should only support EC curves that CAs can issue, and error otherwise. According to 6.1.5 in the CA [BRs](https://cabforum.org/baseline-requirements-documents/) this is P-256, P-384, and P-521.